core/sql/generator/GenRelScan.cpp

```diff
@@ -880,6 +884,12 @@ short FileScan::codeGenForHive(Generator * generator)
   } // for (ii = 0; ii < hdfsVals; ii++)
     
 
+  UInt32 hiveScanMode = CmpCommon::getDefaultLong(HIVE_SCAN_SPECIAL_MODE);
+  //enhance pCode to handle this mode in the future
+  //this is for JIRA 1920
+  if((hiveScanMode & 2 ) > 0)   //if HIVE_SCAN_SPECIAL_MODE is 2, disable pCode
+    exp_gen->setPCodeMode(ex_expr::PCODE_NONE);
+
```

Comment:
This will disable pcode for the whole expression and all conversions.
Better place to disable it is in ex_conv_clause::pcodeGenererate in
file exp/ExpPCodeClauseGen.cpp.
That code should check for conv_to_null_when_error flag in ex_conv_clause
and disable pcode just for that clause.